### PR TITLE
[SPARK-40113][SQL] Reactor ParquetScanBuilder DataSourceV2 interface implementations

### PR DESCRIPTION
**Pull request by yabola**
### What changes were proposed in this pull request?
This PR  unifies `ParquetScanBuilder` in `FileScanBuilder` interface implementations like `Orc`,`Avro`....

### Why are the changes needed?
Unify the logic of the code and make code clearer. 
`FileScanBuilder#pushDataFilters` should return qualified data filter as `OrcScanBuilder`,`AvroScanBuilder`. However `ParquetScanBuilder` did not follow this specification which may lead to bug in the future and difficult to maintain.

### Does this PR introduce any user-facing change?
no

### How was this patch tested?
existing tests

### CODE REVIEW

**Comment by yabola** [comment]
sorry, It's wrong

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/parquet/ParquetScanBuilder.scala [hunk 1]
```diff
@@ -46,7 +45,13 @@ case class ParquetScanBuilder(
     sparkSession.sessionState.newHadoopConfWithOptions(caseSensitiveMap)
   }
 
-  lazy val pushedParquetFilters = {
+  private var finalSchema = new StructType()
+
+  private var pushedAggregations = Option.empty[Aggregation]
+
+  override protected val supportsNestedSchemaPruning: Boolean = true
+
```
Comment (yabola) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/parquet/ParquetScanBuilder.scala:53]: I did not modify this part above, just implemented `FileScanBuilder#pushDataFilters` for ParquetScanBuilder

**Comment by beliefer** [comment]
This PR looks reasonable, but I'm not sure. ping @huaxingao cc @cloud-fan

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/parquet/ParquetScanBuilder.scala [hunk 2]
```diff
@@ -68,25 +73,12 @@ case class ParquetScanBuilder(
         // The rebase mode doesn't matter here because the filters are used to determine
         // whether they is convertible.
         RebaseSpec(LegacyBehaviorPolicy.CORRECTED))
-      parquetFilters.convertibleFilters(pushedDataFilters).toArray
+      parquetFilters.convertibleFilters(dataFilters).toArray
     } else {
       Array.empty[Filter]
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/parquet/ParquetScanBuilder.scala:78]: shouldn't we always return the input `dataFilters`?

Comment (yabola) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/parquet/ParquetScanBuilder.scala:78]: @cloud-fan `FileScanBuilder#pushDataFilters` should return the filters currently supported by the datasource. I refer to the implementation of `OrcScanBuilder` and `JsonScanBuilder`. And this is consistent with the previous parquet implementation logic.

`FileScanBuilder#pushFilters` should always return `dataFilters`.

**Comment by yabola** [comment]
@huaxingao If you have time, please take a look~ I think it's because of some old code, this part is not fully aligned with the new implementation.

**Comment by huaxingao** [comment]
The change looks good to me.

**Review by huaxingao — APPROVED** [review]

**Comment by yabola** [comment]
@cloud-fan cc~

**Comment by yabola** [comment]
@gengliangwang Please also take a look as you are the author of the `ParquetScanBuilder` in the first version~

**Review by cloud-fan — APPROVED** [review]

**Comment by huaxingao** [comment]
Merged to master. Thanks @yabola

**Comment by yabola** [comment]
@cloud-fan @huaxingao Thanks for your review!

**Pull request closed without merging**